src/Microsoft.AspNet.Hosting/Internal/HostingApplicationFeature.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Hosting.Internal
+{
+    public class HostingApplicationFeature
+    {
+        public HostingApplicationFeature(IDisposable logScope, int startTime)
```

Comment:
Is this feature's sole purpose to capture the lifetime of the request?  Maybe RequestLifetimeFeature?